src/highlevel.jl

```diff
@@ -165,14 +190,34 @@ barrier_or!(queue::ROCQueue, signals::Vector{ROCSignal}) =
     Runtime.launch_barrier!(HSA.BarrierOrPacket, queue, signals)
 
 """
-    active_kernels(queue) -> Vector{ROCKernelSignal}
+    active_kernels() -> Vector{ROCKernelSignal}
+    active_kernels(queue::ROCQueue) -> Vector{ROCKernelSignal}
 
 Returns the set of actively-executing kernels on `queue`.
 """
 function active_kernels(queue::ROCQueue=default_queue())
-    lock(Runtime.RT_LOCK) do
-        copy(Runtime._active_kernels[queue])
+    lock(queue.lock) do
+        copy(queue.active_kernels)
+    end
+end
+
+"""
+    synchronize()
+    synchronize(queue::ROCQueue)
+
+Blocks until all kernels currently executing on `queue` have completed. When
+`queue` is unspecified, the default queue is used.
+"""
+function synchronize(queue::ROCQueue=default_queue())
```

Comment:
The one thing that is a tad bit annoying is the name conflicts. But I guess we have to live with that

Noticed that in https://github.com/JuliaGPU/KernelAbstractions.jl/pull/317